source/Modules/_io/TextIO.cpp

```diff
@@ -101,3 +101,154 @@ class TextIO { // 658
 
 	AlifIOState* state{};
 };
+
+
+static AlifIntT _io_TextIOWrapper___init___impl(TextIO* self, AlifObject* buffer,
+	const char* encoding, AlifObject* errors,
+	const char* newline, AlifIntT line_buffering,
+	AlifIntT write_through) { // 1088
+	AlifObject* raw, * codeAlifInfo = nullptr;
+	AlifObject* res{};
+	AlifIntT r{};
+
+	self->ok = 0;
+	self->detached = 0;
+
+	if (encoding == nullptr) {
+		AlifInterpreter* interp = alifInterpreter_get();
+		if (alifInterpreter_getConfig(interp)->warnDefaultEncoding) {
+			//if (alifErr_warnEx(_alifExcEncodingWarning_,
+				//"'encoding' argument not specified", 1)) {
+				//return -1;
+			//}
+		}
+	}
+
+	if (errors == ALIF_NONE) {
+		errors = &ALIF_ID(strict);
+	}
+	else if (!ALIFUSTR_CHECK(errors)) {
+
+		//alifErr_format(
+			//_alifExcTypeError_,
+			//"TextIOWrapper() argument 'errors' must be str or None, not %.50s",
+			//ALIF_TYPE(errors)->name);
+		return -1;
+	}
+	else if (io_checkErrors(errors)) {
+		return -1;
+	}
+	const char* errorsStr = _alifUStr_asUTF8NoNUL(errors);
+	if (errorsStr == nullptr) {
+		return -1;
+	}
+
+	if (validate_newLine(newline) < 0) {
+		return -1;
+	}
+
+	ALIF_CLEAR(self->buffer);
+	ALIF_CLEAR(self->encoding);
+	ALIF_CLEAR(self->encoder);
+	ALIF_CLEAR(self->decoder);
+	ALIF_CLEAR(self->readnl);
+	ALIF_CLEAR(self->decoded_chars);
+	ALIF_CLEAR(self->pendingBytes);
+	ALIF_CLEAR(self->snapshot);
+	ALIF_CLEAR(self->errors);
+	ALIF_CLEAR(self->raw);
+	self->decodedCharsUsed = 0;
+	self->pendingBytesCount = 0;
+	self->encodeFunc = nullptr;
+	self->b2cratIO = 0.0;
+
```

Comment:
if (encoding == nullptr and _alifDureRun.preconfig.utf8Mode) {
        self->encoding = &ALIF_STR(utf_8);
    }